client/python/lib/ovmsclient/tfs_compat/grpc/tensors.py

```diff
@@ -63,43 +96,70 @@ def make_tensor_proto(values, dtype=None, shape=None):
         >>> tensor_proto = make_tensor_proto(data)
         >>> print(tensor_proto)
     '''
-
-    raise NotImplementedError
-
-def _as_numpy_dtype(tensor_dtype):
-    if tensor_dtype == DataType.DT_INVALID:
-        raise ValueError("Tensor data type invalid")
-    elif tensor_dtype == DataType.DT_FLOAT:
-        return np.float32
-    elif tensor_dtype == DataType.DT_DOUBLE:
-        return np.float64
-    elif tensor_dtype == DataType.DT_INT32:
-        return np.int32
-    elif tensor_dtype == DataType.DT_UINT8:
-        return np.uint8
-    elif tensor_dtype == DataType.DT_INT16:
-        return np.int16
-    elif tensor_dtype == DataType.DT_INT8:
-        return np.int8
-    elif tensor_dtype == DataType.DT_STRING:
-        return np.str
-    elif tensor_dtype == DataType.DT_COMPLEX64:
-        return np.complex64
-    elif tensor_dtype == DataType.DT_COMPLEX128:
-        return np.complex128
-    elif tensor_dtype == DataType.DT_INT64:
-        return np.int64
-    elif tensor_dtype == DataType.DT_BOOL:
-        return np.bool
-    elif tensor_dtype == DataType.DT_UINT16:
-        return np.uint16
-    elif tensor_dtype == DataType.DT_UINT32:
-        return np.uint32
-    elif tensor_dtype == DataType.DT_UINT64:
-        return np.uint64
+    if shape is None:
+        inferred_shape = []
+    elif _is_shape_valid(shape):
+        inferred_shape = list(shape)
     else:
-        return np.uint32
-
+        raise TypeError(f'shape type should be list or tuple, but is {type(shape).__name__}')
+    
+    tensor_values = values
+    if np.isscalar(tensor_values):
+        tensor_values = [tensor_values]
+
+    if _is_array_like(tensor_values):
+        if dtype is None:
+            tensor_values = np.array(tensor_values)
```

Comment:
Is there a risk an exception would happen here?

Reply:
with _is_array_like we are sure that tensor_values is either type list or np.ndarray.
There are cases such as this ```values = [[(1,2,3)],[(1,2,3)],[(1,2,3)]], dtype = DataType.DT_INT64```
In that scenario everything works, however with ```values = [[(1,2,3)],[(1,2)],[(1,2,3)]]``` we are getting
```ValueError: setting an array element with a sequence.```
I think that np.array() will not raise an exception while we are sure about the type being list/np.ndarray.
However we can expect exceptions while creating np arrays with declared dtype.

Reply:
That error was generated by ```tensor_values = tensor_values.astype(np_dtype)```